cmd/gcp-controller-manager/gcp_config.go

```diff
@@ -85,10 +85,16 @@ func loadGCPConfig(gceConfigPath string) (gcpConfig, error) {
 	if err != nil {
 		return a, fmt.Errorf("creating GCE API client: %v", err)
 	}
+	if gceAPIEndpointOverride != "" {
+		a.Compute.BasePath = fmt.Sprintf("%sprojects/", gceAPIEndpointOverride)
```

Comment:
Why does this have to include "projects/"?

Reply:
I think it's because the default is `const basePath = "https://www.googleapis.com/compute/v1/projects/"` (from https://raw.githubusercontent.com/googleapis/google-api-go-client/master/compute/v1/compute-gen.go), and when setting the flag you probably just want to pass in the api endpoint (like `https://www.googleapis.com/compute/v1/` or `https://www.googleapis.com/compute/staging_v1/`)?

(Copy-pasted from the k8s gce package, if you prefer I can also drop the projects/ and let user supply it )

Reply:
Hmm, yeah, I think the user should supply the entire prefix to simplify things.
And we can add example v1 prod path in help text for the flag.

Reply:
updated